### PR TITLE
Feature/ui roles

### PR DESCRIPTION
**Pull request by tatianaIljina**

### CODE REVIEW

**Review by DamienVStone — CHANGES_REQUESTED** [review]
merge conflicts please

File: API/Tenders.API/Controllers/DistributionGroupController.cs [hunk 1]
```diff
@@ -198,15 +208,15 @@ public ActionResult Patch([FromBody] DistribuitonGroupViewModel value)
...
+            mapInstance.DistributionRegion = region;
             this.quickSearchService.GenerateQuickSearchString(mapInstance);
 
             if (this.Repository.Update(mapInstance))
             {
                 return this.Ok();
             }
 
-            return this.BadRequest();
+            return this.BadRequest("�� ������� �������� ������ ��������, ����������� ������.");
```
Comment (tatianaIljina) [inline API/Tenders.API/Controllers/DistributionGroupController.cs:219]: !!!

File: API/Tenders.API/Controllers/DistributionGroupSmrController.cs [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+                filter["DistributionType"] = "SmrMonitoring";
+            }
+            else
+            {
+                filter.Add("DistributionType", "SmrMonitoring");
+            }
+
+            var data = this.repository.GetList(filterOptions.Skip(), filterOptions.Take(), filterOptions.QuickSearch, true, filter).ToList();
+            var count = this.repository.Count(0, 0, filterOptions.QuickSearch, true, filter);
+            var res = data.Select(r => new DistribuitonGroupViewModel
```
Comment (tatianaIljina) [inline API/Tenders.API/Controllers/DistributionGroupSmrController.cs:149]: select in new method Map()

File: API/Tenders.API/Controllers/DistributionGroupSmrController.cs [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        [HttpPatch("Update")]
+        [Authorize(Roles = "Admin, Settings.DistributionGroupSmr.Update")]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public ActionResult Patch([FromBody] DistribuitonGroupViewModel value)
+        {
+            if (value == null)
+            {
+                return this.NoContent();
```
Comment (tatianaIljina) [inline API/Tenders.API/Controllers/DistributionGroupSmrController.cs:212]: return BadRequest()

File: API/Tenders.API/Controllers/DistributionGroupSmrController.cs [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+            }
+
+            var region = this.distributionRegionRepository.Find(value.DistributionRegion).FirstOrDefault();
+            if (region == null)
+            {
+                return this.NotFound("Region was not found");
+            }
+
+            var mapInstance = this.distributionGroupFactory.MakeInstance();
+            mapInstance.DistributionType = value.DistributionType;
```
Comment (tatianaIljina) [inline API/Tenders.API/Controllers/DistributionGroupSmrController.cs:234]: Map()

File: Sberbank/Tenders.Sberbank.Requesting/Program.cs [hunk 5]
```diff
@@ -0,0 +1,32 @@
+// <copyright file="Program.cs" company="PlaceholderCompany">
+// Copyright (c) PlaceholderCompany. All rights reserved.
+// </copyright>
+
+namespace Tenders.Sberbank.Requesting
```
Comment (tatianaIljina) [inline Sberbank/Tenders.Sberbank.Requesting/Program.cs:5]: check

**Pull request closed without merging**